lib/bumblebee/text.ex

```diff
@@ -305,6 +305,66 @@ defmodule Bumblebee.Text do
   defdelegate text_classification(model_info, tokenizer, opts \\ []),
     to: Bumblebee.Text.TextClassification
 
+  @type text_embedding_input :: String.t()
+  @type text_embedding_output :: %{embedding: Nx.Tensor.t()}
+
+  @doc """
+  Builds serving for text embeddings.
+
+  The serving accepts `t:text_embedding_input/0` and returns
+  `t:text_embedding_output/0`. A list of inputs is also supported.
+
+  ## Options
+
+    * `:compile` - compiles all computations for predefined input shapes
+      during serving initialization. Should be a keyword list with the
+      following keys:
+
+        * `:batch_size` - the maximum batch size of the input. Inputs
+          are optionally padded to always match this batch size
+
+        * `:sequence_length` - the maximum input sequence length. Input
+          sequences are always padded/truncated to match that length
+
+      It is advised to set this option in production and also configure
+      a defn compiler using `:defn_options` to maximally reduce inference
+      time.
+
+    * `:output_attribute` - the attribute of the embedding model output
+    to return.
```

Comment:
We try to use consistent naming across models, so I think `:pooled_state` is a sensible default :)

```suggestion
    * `:output_attribute` - the attribute of the embedding model output
      to return. Defaults to `:pooled_state`
```

Also nitpick: let's move this as the first option in the list, since compile/defn_options are in each serving and they are the most "boring" ones.